core/src/Slay/Prim.hs

```diff
@@ -82,6 +86,18 @@ data PrimCurve g =
       curveDirection :: g Direction
     }
 
+
+-- | circle adt
+data PrimCircle g = Circle
+    { circleColor :: g (Maybe Color)
+    , circleCenterX :: Double
+    , circleCenterY :: Double
+    , circleRadius :: Double
```

Comment:
Don't use `Double` in a primitive please, I don't like floating-point math. Furthermore, a radius can't be negative. I'd use `Natural` here for now, and later I'll be transitioning to `Ratio Natural`.